docs/Users/Tutorials/Testing_Bears.rst

```diff
@@ -0,0 +1,141 @@
+How to use LocalBearTestHelper to test your bears
+=================================================
+
+*coala* has an awesome testing framework so you can write tests for bears
+with ease.
+
+You can use the following to test your bears:
+
+- ``LocalBearTestHelper.check_validity``
+- ``LocalBearTestHelper.check_results``
+- ``verify_local_bears``
+
+Understanding through examples
+------------------------------
+
+Let us understand how to write tests for ``tooManyLinesBear`` in ``some_dir``.
+``tooManyLinesBear`` checks if a file has less than or equal to
+``max_number_of_lines`` lines. ``max_number_of_lines`` by default is 10.
+
+.. code::
+
+    from coalib.results.Result import Result
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class tooManyLinesBear(LocalBear):
+
+        def run(file,
+                filename,
+                max_number_of_lines: int=10):
+            """
+            Detects if a file has more than "max_number_of_lines" lines
+
+            :param max_number_of_lines    Maximum number of lines to be
+                                          allowed for a file. Default is 10.
+            """
+
+            if(len(file)>max_number_of_lines):
+                yield Result(self, "Too many lines")
+
+**EXAMPLE 1** using ``verify_local_bear``
+
+.. code::
+
+    from bears.some_dir.tooManyLinesBear import tooManyLinesBear
+    from tests.LocalBearTestHelper import verify_local_bear
+
+    good_file = '1\n2\n3\n4\n'.splitlines()
+    bad_file = '1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n'.splitlines()
+
+    tooManyLinesBearTest = verify_local_bear(tooManyLinesBear,
+                                             valid_files=(good_file,),
+                                             invalid_files=(bad_file,))
```

Comment:
we can subtly pass settings here as well to let users know that settings can be changed in `verify_local_bear`